packages/lesswrong/components/ultraFeed/UltraFeedHistoryDialog.tsx

```diff
@@ -0,0 +1,255 @@
+    },
+    '& svg': {
+      display: 'block',
+      strokeWidth: 4.5,
+    },
+  },
+  contentPadding: {
+    padding: 16,
+  },
+  contentContainer: {
+    marginTop: 16,
+    marginLeft: 16,
+    marginRight: 16,
+    maxWidth: SECTION_WIDTH,
+    width: '100%',
+  },
+}));
+
+const UltraFeedHistoryDialog = ({ onClose }: { onClose: () => void }) => {
+  const classes = useStyles(styles);
+  const { settings } = useUltraFeedSettings();
+  const { captureEvent } = useTracking();
+  const [activeTab, setActiveTab] = useState<'history'|'bookmarks'>('history');
+  
+  const setHistory = useCallback(() => {
+    if (activeTab !== 'history') {
+      captureEvent('ultraFeedHistoryTabSwitched', { from: activeTab, to: 'history' });
+      setActiveTab('history');
+    }
+  }, [activeTab, captureEvent]);
+  
+  const setBookmarks = useCallback(() => {
+    if (activeTab !== 'bookmarks') {
+      captureEvent('ultraFeedHistoryTabSwitched', { from: activeTab, to: 'bookmarks' });
+      setActiveTab('bookmarks');
+    }
+  }, [activeTab, captureEvent]);
+
+  const handleClose = useCallback(() => {
+    captureEvent('ultraFeedHistoryDialogClosed', { activeTab });
+    onClose();
+  }, [activeTab, captureEvent, onClose]);
+
+  // Track when dialog is opened
+  useEffect(() => {
+    captureEvent('ultraFeedHistoryDialogOpened', { initialTab: activeTab });
+  }, [captureEvent, activeTab]);
+
+  useDisableBodyScroll();
+
+  const { feedCommentThread, feedPost, feedSpotlight } = createUltraFeedRenderers({ settings });
+
+  return (
+    <AnalyticsContext pageSectionContext="ultraFeedHistoryDialog">
+      <UltraFeedObserverProvider incognitoMode={true}>
+        <OverflowNavObserverProvider>
+          <LWDialog
+            open={true}
+            onClose={handleClose}
+            className={classes.modalWrapper}
+            paperClassName={classes.dialogPaper}
+          >
+            <DialogContent className={classes.dialogContent}>
+              <div className={classes.header}>
+                <div className={classes.headerLeft}>
+                  <ForumIcon
+                    icon="ArrowLeft"
+                    className={classes.closeButton}
+                    onClick={handleClose}
+                  />
+                </div>
+                <div className={classes.headerTabs}>
+                  <div
+                    className={`${classes.tabButton} ${activeTab === 'history' ? classes.tabButtonActive : classes.tabButtonInactive}`}
+                    onClick={setHistory}
+                  >
+                    <div className={classes.tabLabel}>
+                      History
+                      {activeTab === 'history' && (
+                        <div className={classes.tabUnderline} />
+                      )}
+                    </div>
+                  </div>
+                  <div
+                    className={`${classes.tabButton} ${activeTab === 'bookmarks' ? classes.tabButtonActive : classes.tabButtonInactive}`}
```

Comment:
This is not how we compose classnames - maybe it works but I don't want to have two different conventions in the codebase.  (I'm not going to call every instance out individually, but please do a pass and clean them up)